### ISSUE TITLE
Package updates for maestral

### ISSUE THREAD

**Issue by SFrijters** [NixOS/nixpkgs#87796]
**Project description**

Maestral is already in nixpkgs at v0.6.4, but at this time a new version v1.0.2 is available.
It looks like the update is not completely trivial however.

* The gui has been split off into two new libraries: maestral-qt (for Linux) and maestral-cocoa (for darwin). These are not yet present in nixpkgs.
* The lower bound on the `dropbox` Python library has been raised to 10, and that PR #87315 is stalled because the current 0.6.4 version of maestral doesn't like the new dropbox. So I think they have to be updated in tandem?
* If I switch on the withGui flag for the current package, the build aborts because makeQtWrapper is missing.

@peterhoeg I can take a poke at this at some point, but I would probably need some help. Also, what is the policy on updating multiple packages together?

**Metadata**

* homepage URL: https://github.com/SamSchott/maestral
* source URL:
* license: mit
* platforms: unix

**Comment by peterhoeg** [comment]
I'm running 1.0.2 here (without the gui though) so the attached patch would be a good starting point.

Would be great if you could get everything working!

diff --git a/pkgs/applications/networking/maestral/default.nix b/pkgs/applications/networking/maestral/default.nix
index fe611892ae6..630d9e2e5f1 100644
--- a/pkgs/applications/networking/maestral/default.nix
+++ b/pkgs/applications/networking/maestral/default.nix
@@ -8,7 +8,7 @@
 
 python3.pkgs.buildPythonApplication rec {
   pname = "maestral${lib.optionalString withGui "-gui"}";
-  version = "0.6.4";
+  version = "1.0.2";
 
   disabled = python3.pkgs.pythonOlder "3.6";
 
@@ -16,7 +16,7 @@ python3.pkgs.buildPythonApplication rec {
     owner = "SamSchott";
     repo = "maestral-dropbox";
     rev = "v${version}";
-    sha256 = "06psbjd205qj6vx7k7ywhclz1bs5zp43xnf275hf0k80h3b064sn";
+    sha256 = "16wmw7ximdlbv8vzc57xjrki0df29mps5r9xwx7iifbvp9n0m0cl";
   };
 
   propagatedBuildInputs = with python3.pkgs; [
diff --git a/pkgs/development/python-modules/dropbox/default.nix b/pkgs/development/python-modules/dropbox/default.nix
index 3ca9d59e27f..83ecf7f29dd 100644
--- a/pkgs/development/python-modules/dropbox/default.nix
+++ b/pkgs/development/python-modules/dropbox/default.nix
@@ -3,11 +3,11 @@
 
 buildPythonPackage rec {
   pname = "dropbox";
-  version = "9.5.0";
+  version = "10.0.0";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "0iz9hg1j7q9chka6fyzgpzqg2v4nbjx61xfvn9ixprxrdhvhr2hi";
+    sha256 = "0h20lb9krih9i7xdanx275q1xnqx1i623qzvans4iahi2a5iiikj";
   };
 
   # Set DROPBOX_TOKEN environment variable to a valid token.

**Comment by peterhoeg** [comment]
>  I can take a poke at this at some point, but I would probably need some help. 

Sure, just ping me.

> Also, what is the policy on updating multiple packages together?

As long as they are related, it's not a problem at all. They do need to be in separate commits though.

**Comment by SFrijters** [comment]
I was shuffling things around a bit, but I ran into a problem which already exists using your patch only:

```
$ nix-build /home/frijters/repos/nixpkgs/ -A maestral
/nix/store/fpn6bahmcyzdajyx12isxcb3scya32aj-maestral-1.0.2
$ /home/frijters/repos/nixpkgs/result/bin/maestral start
Starting Maestral...Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'maestral'
```
Have you seen this before? The current 0.6.4 version doesn't give this error for me (although the daemon doesn't work anyway - this is the reason why I wanted to try and upgrade).

It looks like the spawned daemon doesn't have the correct module paths set? If I run it with `--foreground` it works.

**Comment by SFrijters** [comment]
This is disgusting but it seems to work? Is there a nicer way to express this?

```
  makeWrapperArgs = [
    "--prefix" "PYTHONPATH" ":" "${stdenv.lib.concatStringsSep ":" (map (p: p + "/lib/${python.libPrefix}/site-packages") (python.pkgs.requiredPythonModules pbi))}"
  ];
```
where `pbi` is a let binding that's also used for `propagatedBuildInputs`.

**Comment by SFrijters** [comment]
@peterhoeg Could you take a look at #88698 ?

**Comment by SFrijters** [comment]
This was closed by #88698.

**Comment by peterhoeg** [comment]
Thanks for your persistence @SFrijters !
